File: workspace_java/JAVA0415/src/switch_/Test2.java
package switch_;

public class Test2 {

	public static void main(String[] args) {
		// 점수(score)에 따른 학점 계산을 switch 문으로 수행할 경우
		// case 문에서 값만 지정 가능하며 범위 지정 불가능하므로
		// 각각의 점수를 모두 지정해야 함
		int score = 85;
		
		switch(score) {
			case 100 : System.out.println("A학점"); break;
			case 99 : System.out.println("A학점"); break;
			case 98 : System.out.println("A학점"); break;
			// ... 생략 ...
			case 89 : System.out.println("B학점"); break;
			case 88 : System.out.println("B학점"); break;
			// ... 생략 ...
			case 1 : System.out.println("F학점"); break;
			case 0 : System.out.println("F학점"); break;
		}
		
		System.out.println("---------------------------------");

		/*
		 * 위의 switch 문 코드를 줄이기 위해 연산식을 활용
		 * => 각 학점에 해당하는 점수의 공통점을 추출
		 * A 학점 : 90점대와 100점(= 10의 자리 숫자가 9 또는 10)
		 * B 학점 : 80점대(= 10의 자리 숫자가 8)
		 * C 학점 : 70점대(= 10의 자리 숫자가 7)
		 * D 학점 : 60점대(= 10의 자리 숫자가 6)
		 * F 학점 : 나머지 점수
		 * => 점수에서 10의 자리 값을 추출하여 switch 문에 전달
		 */
		
		score = 105;
		
		// 문제점 : 101 ~ 109 점수도 A학점으로 취급됨
		// => 해결책 : if문을 사용하여 0 ~ 100점 사이일 때만 학점 계산
		
		if(score >= 0 && score <= 100) {
			switch(score / 10) {
				case 10 :
				case 9 : System.out.println(score + " : A 학점!"); break;
				case 8 : System.out.println(score + " : B 학점!"); break;
				case 7 : System.out.println(score + " : C 학점!"); break;
				case 6 : System.out.println(score + " : D 학점!"); break;
				default : System.out.println(score + " : F 학점!");
			}
			
		} else {
			System.out.println(score + " : 점수 입력 오류!");
		}
		
		
		
	}

}

File: Backend/workspace_backend/StudyJavascript/src/main/webapp/javascript7_0421_object/test2.js
/*
3) 클래스(Class) 선언 방식
   - 자바에서 정의하는 방식과 거의 동일한 문법 사용(ES6 이상부터 사용 가능)
   
   [ 기본 문법 ]
   class 클래스명 {
		constructor([파라미터...]) {
			this.프로퍼티명1 = 파라미터1; 
			this.프로퍼티명2 = 파라미터2,
			...생략...
			this.프로퍼티명n = 파라미터n
		}
	
		메서드명([파라미터...]) {
			메서드 호출 시 실행할 코드...
		}
   }
   
   => 객체 생성 및 접근 방법은 1, 2번 방법과 동일함
*/

class Car {
	constructor(modelName, color) { // new Car(modelName, color) 형태로 객체 생성 필요함
		this.modelName = modelName;
		this.color = color;
	}
	
	engineStart() {
		console.log("차 시동 걸기!");
	}
}

// ========================================================
function test2() {
	// Car 클래스의 인스턴스 생성
	const car = new Car("SONATA", "RED");
	
	// Car 인스턴스의 프로퍼티값 출력
	console.log("car 객체의 modelName 속성값 : " + car.modelName);
	console.log(`car 객체의 color 속성값 : ${car.color}`);
	
	// Car 인스턴스의 engineStart() 메서드 호출
	car.engineStart();
	// ------------------------------
	// 동일한 Car 클래스를 바탕으로 두번째 Car 인스턴스(객체) 생성하기
	const car2 = new Car(); // 위의 Car 객체와는 또 다른 객체가 되어 서로 다른 값을 갖게 된다!
	// => 주의! 생성자가 Car(modelName, color) 형태로 호출되어야 하지만
	//    생성자 파라미터로 아무것도 전달하지 않으면 속성값이 없게 되므로 undefined 가 된다! 
	console.log("car2 객체의 modelName 속성값 : " + car2.modelName); // undefined
	console.log(`car2 객체의 color 속성값 : ${car2.color}`); // undefined
	car2.engineStart();
}























